popsycle/synthetic.py

```diff
@@ -3758,133 +3773,151 @@ def qnfw(p, con=5, logp=False):
-
-    for key in key_list:
-        keys = pd.DataFrame(np.array(no_pbh_hdf5_file['{}'.format(key)]))
-        vars()[key+'_max_l'] = np.amax(keys.T[11])
-        vars()[key+'_min_l'] = np.amin(keys.T[11])
-        vars()[key+'_max_b'] = np.amax(keys.T[10])
-        vars()[key+'_min_b'] = np.amin(keys.T[10])
-        vars()[key+'_mask'] = (pbh_data['glon'] >= vars()[key+'_min_l']) & (pbh_data['glon'] <= vars()[key+'_max_l']) & (pbh_data['glat'] >= vars()[key+'_min_b']) & (pbh_data['glat'] <= vars()[key+'_max_b'])
-        vars()['pbh_'+key] = (pbh_data[vars()[key+'_mask']]).T
-        vars()['pbh_'+key].reset_index(drop=True, inplace=True)
-        vars()['full_'+key]=pd.concat([keys, vars()['pbh_'+key]], axis=1)
+    pbh_hdf5_file = h5py.File(output_root2 + '.h5', 'w')
+
+    #Calculate the maximum and minimum l and b values for each dataset in the no PBH file, 
+    #so that we can determine which datasets to correctly add the PBHs.
+    lat_bin_values = lat_bin.values
+    long_bin_values = long_bin.values
+    lat_long_list = []
+    for idx in range(len(lat_bin_values)-1):
+        max_l = lat_bin_values[idx+1]
+        min_l = lat_bin_values[idx]
+        for idx2 in range(len(long_bin_values)-1):
+            max_b = long_bin_values[idx2+1]
+            min_b = long_bin_values[idx2]
+            lat_long_list.append((min_l[0], max_l[0], min_b[0], max_b[0]))
+
+    #Appending the PBH data to the no PBH data and writing to the new .h5 file.
+    for idx, key in enumerate(key_list):
+        data = pd.DataFrame(no_pbh_hdf5_file[key][:])
+        min_l, max_l, min_b, max_b = lat_long_list[idx]
+        mask = (pbh_data['glon'] >= min_b) & (pbh_data['glon'] <= max_b) & (pbh_data['glat'] >= min_l) & (pbh_data['glat'] <= max_l)
+        pbh_key = pbh_data[mask].T
+        pbh_key.reset_index(drop=True, inplace=True)
+        full_key = pd.concat([data, pbh_key], axis=1)
+        d_=pbh_hdf5_file.create_dataset(key, (full_key.shape[0], full_key.shape[1]), data=full_key)
+    d_lat = pbh_hdf5_file.create_dataset('lat_bin_edges', (len(lat_bin), 1), data=lat_bin)
+    d_long = pbh_hdf5_file.create_dataset('long_bin_edges', (len(lat_bin), 1), data=long_bin)
     no_pbh_hdf5_file.close()
+    pbh_hdf5_file.close()
 
-    #Creating final .h5 file with PBHs injected
-    f = h5py.File(output_root2 + '.h5', 'w')
-
-    for key in key_list:
-        vars()['d_'+key]=f.create_dataset(key, (vars()['full_'+key].shape[0], vars()['full_'+key].shape[1]), data=vars()['full_'+key])
-
-    d_lat = f.create_dataset('lat_bin_edges', (len(lat_bin), 1), data=lat_bin)
-    d_long = f.create_dataset('long_bin_edges', (len(lat_bin), 1), data=long_bin)
-    f.close()
+    t1 = time.time()
+    print('Total runtime: {0:f} s'.format(t1 - t0))
 
     return
 
-def generate_pbh_config_file(config_filename, fdm, pbh_mass, r_max, c, r_vir):
+def generate_pbh_config_file(config_filename, fdm, pbh_mass, r_max, c, r_vir, inner_slope, v_esc):
```

Comment:
Either set default values in this definition, or remove the language about default values in the docstring below.